### PR TITLE
✨ Add `rule_psr` to RUM resource

### PR DESCRIPTION
**Pull request by liywjl**
As we are now passing `rule_psr` via the browser SDK to backend, it makes sense to add to here

`psr` = `Percentage Sample Rate`

### CODE REVIEW

File: schemas/rum/resource-schema.json [hunk 1]
```diff
@@ -252,6 +252,12 @@
               "description": "trace identifier in decimal format",
               "pattern": "^[0-9]+$",
               "readOnly": true
+            },
+            "rule_psr": {
+              "type": "number",
+              "description": "sample rate in decimal format",
+              "pattern": "^[0-9]+$",
```
Comment (bcaudan) [inline schemas/rum/resource-schema.json]: **❓ question:** ‏I guess we are not limited to integers, no?

File: schemas/rum/resource-schema.json [hunk 2]
```diff
@@ -252,6 +252,12 @@
               "description": "trace identifier in decimal format",
               "pattern": "^[0-9]+$",
               "readOnly": true
+            },
+            "rule_psr": {
+              "type": "number",
+              "description": "sample rate in integer format",
+              "pattern": "^[1-9]$|^[1-9][0-9]$|^(100)$",
```
Comment (BenoitZugmeyer) [inline schemas/rum/resource-schema.json]: **💬 suggestion:** ‏remove the `pattern` property here

I don't think the `pattern` keyword applies on `number` ([in the docs](https://json-schema.org/understanding-json-schema/reference/string.html#pattern) I only see it mentioned on `string` or object keys). Even if it does, I don't think checking the number format is useful here. Using a [range](https://json-schema.org/understanding-json-schema/reference/numeric.html?highlight=number#range) would be more appropriate.

File: schemas/rum/resource-schema.json (hunk 1 above)
Comment (bcaudan) [reply schemas/rum/resource-schema.json]: Looking at your change, I was probably not clear enough, I am thinking that we could provide values like `12,34` for example, so we should not limit this field to integer value.

Comment (liywjl) [reply schemas/rum/resource-schema.json]: It should be clearer using the [range](https://json-schema.org/understanding-json-schema/reference/numeric.html?highlight=number#range) keys mentioned by @BenoitZugmeyer

**Review by ncreated — COMMENTED** [review]
What does "PSR Rule" stand for? 🤔💭

**Review by xgouchet — APPROVED** [review]

**Comment by liywjl** [comment]
> What does "PSR Rule" stand for? 🤔💭

I added the definition to the description

File: schemas/rum/resource-schema.json [hunk 3]
```diff
@@ -242,6 +242,13 @@
               "description": "trace identifier in decimal format",
               "pattern": "^[0-9]+$",
               "readOnly": true
+            },
+            "rule_psr": {
+              "type": "number",
+              "description": "sample rate in integer format",
```
Comment (xgouchet) [inline schemas/rum/resource-schema.json]: Maybe we need to be more precise as to which sampling rate this refer to. We have (at least) three: RUM session, RUM premium session and Tracing

File: schemas/rum/resource-schema.json [hunk 4]
```diff
@@ -242,6 +242,13 @@
               "description": "trace identifier in decimal format",
               "pattern": "^[0-9]+$",
               "readOnly": true
+            },
+            "rule_psr": {
+              "type": "number",
+              "description": "tracing sample rate in integer format",
+              "minimum": 0,
+              "exclusiveMaximum": 100,
```
Comment (bcaudan) [inline schemas/rum/resource-schema.json]: **💬 suggestion:** ‏Should it not be 1 then?

File: schemas/rum/resource-schema.json [hunk 5]
```diff
@@ -242,6 +242,13 @@
               "description": "trace identifier in decimal format",
               "pattern": "^[0-9]+$",
               "readOnly": true
+            },
+            "rule_psr": {
+              "type": "number",
+              "description": "tracing sample rate in decimal format",
+              "minimum": 0,
+              "exclusiveMaximum": 1,
```
Comment (BenoitZugmeyer) [inline schemas/rum/resource-schema.json]: **❓ question:** ‏I think this should be inclusive?


**💬 suggestion:** ‏
```suggestion
              "maximum": 1,
```

**Review by bcaudan — APPROVED** [review]

**Review by BenoitZugmeyer — APPROVED** [review]

**Pull request merged**